engine/src/main/java/org/terasology/engine/logic/location/LocationChangedEvent.java

```diff
@@ -2,67 +2,56 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.engine.logic.location;
 
+import com.google.common.base.Preconditions;
 import org.joml.Quaternionf;
 import org.joml.Quaternionfc;
 import org.joml.Vector3f;
 import org.joml.Vector3fc;
 import org.terasology.engine.entitySystem.event.Event;
 
 public class LocationChangedEvent implements Event {
-    public final LocationComponent component;
-    public final Vector3fc oldPosition;
-    public final Quaternionfc oldRotation;
-    public final Vector3fc newPosition;
-    public final Quaternionfc newRotation;
+    protected Vector3f lastPosition = new Vector3f();
+    protected Quaternionf lastRotation = new Quaternionf();
+    protected Vector3f currentPosition = new Vector3f();
+    protected Quaternionf currentRotation = new Quaternionf();
 
-    public LocationChangedEvent(LocationComponent newLocation) {
-        this(newLocation, newLocation.position, newLocation.rotation, newLocation.position, newLocation.rotation);
+    public LocationChangedEvent() {
     }
 
-    public LocationChangedEvent(LocationComponent newLocation, Vector3f oPosition) {
-        this(newLocation, oPosition, newLocation.rotation, newLocation.position, newLocation.rotation);
-    }
-
-    public LocationChangedEvent(LocationComponent newLocation, Vector3f oPosition, Vector3f nPosition) {
-        this(newLocation, oPosition, newLocation.rotation, nPosition, newLocation.rotation);
-    }
+    public LocationChangedEvent(Vector3fc lastPosition, Quaternionfc lastRotation, Vector3fc currentPosition, Quaternionfc currentRotation) {
+        Preconditions.checkNotNull(lastPosition);
+        Preconditions.checkNotNull(lastRotation);
+        Preconditions.checkNotNull(currentPosition);
+        Preconditions.checkNotNull(currentRotation);
 
-    public LocationChangedEvent(LocationComponent newLocation, Quaternionfc oRotation) {
-        this(newLocation, newLocation.position, oRotation, newLocation.position, newLocation.rotation);
+        this.lastPosition.set(lastPosition);
+        this.lastRotation.set(lastRotation);
+        this.currentPosition.set(currentPosition);
+        this.currentRotation.set(currentRotation);
     }
 
-    public LocationChangedEvent(LocationComponent newLocation, Quaternionfc oRotation, Quaternionfc nRotation) {
-        this(newLocation, newLocation.position, oRotation, newLocation.position, nRotation);
+    public Vector3fc lastPosition() {
```

Comment:
I think `last` and `current` works well here as naming. In `BeforeAfterEvent` we use `old` and `new` - was just wondering whether we should be consistent here :thinking: